include/arkode/arkode.h

```diff
@@ -73,6 +73,7 @@ extern "C" {
 #define ARK_SUCCESS      0
 #define ARK_TSTOP_RETURN 1
 #define ARK_ROOT_RETURN  2
+#define ARK_RETRY_STEP   3
```

Comment:
What is this value for? From this name it seems like an internal flag value rather than something the user should be aware of.

Reply:
It is internal to ARKODE, as part of the ARKODE / stepper interface.  Essentially, if temporal adaptivity has selected a given step size, but an STS method is not allowed to take enough stages to make that step stable, we want to abort the step attempt (after resetting `ark_mem->eta`) and have ARKODE retry the step with the smaller step size.

Perhaps we should add a new value to `arkode_impl.h` instead and use that?